internal/service/backupcompliancepolicy/resource_backup_compliance_policy_test.go

```diff
@@ -116,6 +117,41 @@ func TestAccBackupCompliancePolicy_withoutRestoreWindowDays(t *testing.T) {
 	})
 }
 
+func TestAccBackupCompliancePolicy_UpdateSetsAllAttributes(t *testing.T) {
+	var (
+		orgID          = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName    = acc.RandomProjectName() // No ProjectIDExecution to avoid conflicts with backup compliance policy
+		projectOwnerID = os.Getenv("MONGODB_ATLAS_PROJECT_OWNER_ID")
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: configBasicWithOptionalAttributesWithNonDefaultValues(projectName, orgID, projectOwnerID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "authorized_user_first_name", "First"),
+					resource.TestCheckResourceAttr(resourceName, "authorized_user_last_name", "Last"),
+					resource.TestCheckResourceAttr(resourceName, "pit_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "encryption_at_rest_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "copy_protection_enabled", "true"),
+				),
+			},
+			{
+				Config: configBasicWithOptionalAttributesWithNonDefaultValues(projectName, orgID, projectOwnerID),
```

Comment:
Since we have fixed the update operation, should we provide a modified config in the test so that the update operation is invoked?

Reply:
In this case is important to check the exact same config with an Optional attribute with a non default value. What was happening in the update is that if an Optional attribute wasn't modified, the Optional attributes went back to the default value. 
Other updates of fields were being tested already

Reply:
My question is that if the second step provides the same configuration as in first step, terraform will not trigger an update operation. I understand we want to preserve the same value (non default) for an optional attribute, but we still have to modify another attribute so that the update is actually called.